src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -196,15 +370,24 @@ def create_if_needed(self, app_name):
     def create(self):
         self.location = validate_environment_location(self.cmd, self.location)
         register_provider_if_needed(self.cmd, LOG_ANALYTICS_RP)
-        env = create_managed_environment(
-            self.cmd,
-            self.name,
-            location=self.location,
-            resource_group_name=self.resource_group.name,
-            logs_key=self.logs_key,
-            logs_customer_id=self.logs_customer_id,
-            disable_warnings=True,
-        )
+        if self.is_connected_environment_type():
```

Comment:
suggestion: when creating env, if there is custom location provided, then you need to create connected env. otherwise, create managed env.